pretty/src/pane.rs

```diff
@@ -172,6 +190,14 @@ where
                     row: cursor_region.pos.row,
                 };
                 doc.pretty_print(width, self, doc_pos)?;
+
+                if let CursorVis::Show = cursor_visibility {
+                    let region = Region {
```

Comment:
This needs to be an argument to `pretty_print`, not `Pane`. The way it's done here won't work when there are multiple levels of cursor shading, I don't think.

Reply:
Yeah, certainly once there are multiple levels of shading, it seems like that should be handled by `pretty_print`. I'm not sure what other changes will be required to make the multiple levels work. Want me to move this cursor-shading code into `pretty_print()` now, or put it on the todo list? It will involve adding `CursorVis::Hide` to a lot of test cases.